src/kerberos.c

```diff
@@ -211,14 +211,56 @@ static PyObject *authGSSClientClean(PyObject *self, PyObject *args)
     return Py_BuildValue("i", result);
 }
 
-static PyObject *authGSSClientStep(PyObject *self, PyObject *args)
+static PyObject *buildChannelBindingsStruct(PyObject *self, PyObject *args, PyObject* keywds)
+{
+    int initiator_addrtype = GSS_C_AF_UNSPEC;
+    int acceptor_addrtype = GSS_C_AF_UNSPEC;
+    int initiator_length = 0;
+    int acceptor_length = 0;
+    int application_length = 0;
+    const char *initiator_address = NULL;
+    const char *acceptor_address = NULL;
+    const char *application_data = NULL;
+    int result = 0;
+
+    PyObject *pychan_bindings = NULL;
+    struct gss_channel_bindings_struct *input_chan_bindings;
+	static char *kwlist[] = {"initiator_addrtype", "initiator_address", "accept_addrtype",
+	                        "acceptor_address", "application_data", NULL};
+
+	if (!PyArg_ParseTupleAndKeywords(args, keywds, "|is#is#s#", kwlist,
```

Comment:
Yes, specifically using PyMem_Free, since PyArg_ParseTuple allocated the storage for you. The thing that worries me about using s# is that PyArg_ParseTuple just gives you a pointer to the internal storage of the Python string / unicode object. If that object's ref count falls to zero while the application is still using the binding object you'll have a problem.

The C API docs for python 3.6 make this clear:
```
In general, when a format sets a pointer to a buffer, the buffer is managed by the corresponding Python object, and the buffer shares the lifetime of this object.
```